### ISSUE TITLE
Support screenshot testing with gfx.Display

### ISSUE THREAD

**Issue by Korijn** [pygfx/pygfx#386]
As a follow-up to #380 we also should implement support for gfx.Display in screenshot tests.

**Comment by FirefoxMetzger** [comment]
How would this look like? Currently we can do something like:

```python
disp = gfx.Display()
disp.show(scene)
img = disp.renderer.snapshot()
```

**Comment by Korijn** [comment]
https://github.com/pygfx/pygfx/blob/98f5b3ca25430d7c0bc7fe3feee103057017b4c5/examples/tests/test_examples.py#L82-L84

That second line would need to become a try/catch statement to optionally get the renderer via the `disp` variable.

**Comment by FirefoxMetzger** [comment]
Ah, you mean to modify `test_examples.py` to use `gfx.Display`?

**Comment by Korijn** [comment]
Yes!

**Comment by almarklein** [comment]
> We could also iterate over the module attributes and grab a renderer/display reference,

This may break if an example uses multiple renderers (e.g. to a texture shown in the final scene).

**Comment by Korijn** [comment]
> > We could also iterate over the module attributes and grab a renderer/display reference,
> 
> This may break if an example uses multiple renderers (e.g. to a texture shown in the final scene).

Yeah. Nevermind. Too much magic. :)

**Comment by almarklein** [comment]
Implemented in #384
